privacy_evaluator/attacks/property_inference_attack.py

```diff
@@ -282,22 +252,25 @@ def attack(self):
         feature_extraction_target_model = self.feature_extraction(self.target_model)
 
         # set ratio and size for unbalanced data sets
-        size_set = 1000 #TODO get size of on class of dataset
+        size_set = 1000  # TODO get size of one class of dataset
 
-        #balanced ratio
+        # balanced ratio
         num_elements = int(round(size_set / 2))
-        neg_property_num_elements_per_classes={0: num_elements, 1: num_elements}
+        neg_property_num_elements_per_class = {0: num_elements, 1: num_elements}
 
-        #create balanced shadow classifiers (negation property)
-        shadow_classifiers_neg_property, accuracy_neg_property = self.create_shadow_classifier_from_training_set(neg_property_num_elements_per_classes)
+        # create balanced shadow classifiers (negation property)
+        shadow_classifiers_neg_property = self.create_shadow_classifier_from_training_set(
+            neg_property_num_elements_per_class
+        )
       
-        predictions = dict()
-        #iterate over ratios from 0.55 to 0.95 (means: class 0: 0.45 of all samples, class 1: 0.55 of all samples)
-        #TODO add more
-        for ratio in np.arange(0.55, 1, 0.05):#range(0.55, 0.95, 0.05):
-
-            predictions[round(ratio, 5)] = self.prediction_on_specific_property(feature_extraction_target_model, shadow_classifiers_neg_property,ratio, size_set)
-
-            predictions[round((1-ratio),5)] = self.prediction_on_specific_property(feature_extraction_target_model, shadow_classifiers_neg_property,(1-ratio), size_set)
+        predictions = {}
+        # iterate over ratios from 0.55 to 0.95 (means: class 0: 0.45 of all samples, class 1: 0.55 of all samples)
+        # TODO add more
+        for ratio in np.arange(0.55, 1, 0.05):
```

Comment:
Why do you hard-code the values? And How could you, for example, check for class 0: 90% and class 1 10%? As far as I can see, right now you could only have class 0 10% and class 1 90%, not the other way round.

Reply:
You can check both cases (90,10 & 10,90) because we use ratio and 1-ratio in every iteration. I added a more detailed explanation about this in our current code, so hopefully it is now easier to see in our code. I added hard-code values to the topics to discuss for tomorrow since we were unsure about this in some cases.